### PR TITLE
base/*: Set skip_missing_names_on_install=0 in /etc/yum.conf

### PR DESCRIPTION
**Pull request by chancez**
Ensure yum install fails if a package is missing.

### CODE REVIEW

File: base/Dockerfile.rhel [hunk 1]
```diff
@@ -6,6 +6,7 @@ RUN INSTALL_PKGS=" \
       " && \
     if [ ! -e /usr/bin/yum ]; then ln -s /usr/bin/microdnf /usr/bin/yum; fi && \
     yum install -y --setopt=tsflags=nodocs ${INSTALL_PKGS} && \
-    yum clean all && rm -rf /var/cache/*
+    yum clean all && rm -rf /var/cache/* && \
+    echo 'skip_missing_names_on_install=0' >> /etc/yum.conf
```
Comment (smarterclayton) [inline base/Dockerfile.rhel]: Do we need a leading newline?

Comment (chancez) [reply base/Dockerfile.rhel]: Not that I saw when I tested in the container manually.

**Comment by sdodson** [comment]
/lgtm cancel

> Do we need to set the yum.conf file before the yum install?

Good idea, I think the main goal was to fix downstream images but I think reordering things here as well makes sense.

File: base/Dockerfile [hunk 2]
```diff
@@ -14,7 +14,8 @@ RUN INSTALL_PKGS=" \
     yum install -y ${INSTALL_PKGS} && \
     rpm -V ${INSTALL_PKGS} && \
     yum clean all && \
-    mkdir -p /var/lib/origin
+    mkdir -p /var/lib/origin && \
+    echo 'skip_missing_names_on_install=0' >> /etc/yum.conf
```
Comment (smarterclayton) [inline base/Dockerfile]: Yeah move this up and then nuke `rpm INSTALL_PKGS`

Comment (chancez) [reply base/Dockerfile]: sounds good

**Comment by chancez** [comment]
Updated to move the config option before yum install.

**Pull request merged**